cmd/clusterctl/client/config.go

```diff
@@ -54,6 +55,45 @@ func (c *clusterctlClient) GetProviderComponents(provider string, providerType c
 	return components, nil
 }
 
+func (c *clusterctlClient) GenerateProvider(providerName string, providerType clusterctlv1.ProviderType, options ComponentsOptions) (Components, error) {
+
+	providerName, providerVersion, err := parseProviderName(providerName)
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, errrors.Wrap(err, "failed to parse provider name")
```

Wrapping the errors provides more context when something fails. 
If you can also wrap other errors below that would be great.